src/containers/TriggerTemplate/TriggerTemplate.js

```diff
@@ -0,0 +1,372 @@
+            {intl.formatMessage({
+              id: 'dashboard.triggerTemplate.details',
+              defaultMessage: 'Details'
+            })}
+          </Button>
+          <Button onClick={() => this.handleYAML()} kind="ghost">
+            {intl.formatMessage({
+              id: 'dashboard.triggerTemplate.yaml',
+              defaultMessage: 'YAML'
+            })}
+          </Button>
+        </div>
+
+        {this.state.showYAML && (
+          /* eslint-disable react/no-unknown-property */
+          <div className="textarea">
+            <pre id="yaml">
+              <p className="triggertemplatetext">
+                {jsYaml.dump(triggerTemplate)}
+              </p>
+            </pre>
+          </div>
+        )}
+
+        {this.state.showDetails && (
+          /* eslint-disable react/no-unknown-property */
+          <div className="textarea">
+            <pre id="yaml">
+              <p className="triggertemplatetext">
+                <p>
+                  {intl.formatMessage({
+                    id: 'dashboard.triggerTemplate.creationTime',
+                    defaultMessage: 'Creation time'
+                  })}
+                  : {triggerTemplateDate}
+                </p>
+                <p>
+                  {intl.formatMessage({
+                    id: 'dashboard.triggerTemplate.namespace',
+                    defaultMessage: 'Namespace'
+                  })}
+                  : {triggerTemplateNamespace}
+                </p>
+
+                {triggerTemplateLabels && (
+                  <p>
+                    <>
+                      {intl.formatMessage({
+                        id: 'dashboard.triggerTemplate.labels',
+                        defaultMessage: 'Labels'
+                      })}
+                      : ${triggerTemplateLabels}
+                    </>
+                  </p>
+                )}
+              </p>
+            </pre>
+          </div>
+        )}
+
+        <DashboardTable
+          title="Parameters"
+          headers={headersForParameters}
+          rows={rowsForParameters}
+          selectedNamespace={selectedNamespace}
+          emptyTextAllNamespaces={intl.formatMessage(
+            {
+              id: 'dashboard.emptyState.allNamespaces',
+              defaultMessage: 'No {kind} under any namespace.'
+            },
+            { kind: 'TriggerTemplates' }
+          )}
+          emptyTextSelectedNamespace={intl.formatMessage(
+            {
+              id: 'dashboard.emptyState.selectedNamespace',
+              defaultMessage: 'No {kind} under namespace {selectedNamespace}'
+            },
+            { kind: 'TriggerTemplate', selectedNamespace }
+          )}
+          isSortable
+        />
+
+        {resourcetemplates && (
+          <div className="tableComponent">
+            <DataTable
+              rows={resourcetemplates.map(item => ({
+                id: `${item.metadata.name ||
```

Comment:
not sure what's going on here... guessing this is the hack you referred to. Can you explain what we're trying to do here? We should be able to find a cleaner solution than JSON -> YAML -> string concat -> split, etc.

Reply:
Yeah, so what I'm trying to say is

For the ID, use the metadata.name or metadata.generateName and then add **data** followed by the full json for the object itself

Because we do a map and iterate over id, we can then use it later on in the accordion so that's the content displayed onclick

Reply:
can we use the id to look up the data where it's needed instead of going through this whole dance here?

Reply:
Yeah good suggestion, I think that's indeed nicer than an Accordion actually (I didn't think it existed)

@carlos-logro FYI as you did suggest it and I hadn't realised